### PR TITLE
SDN-3817,MON-3795:Expose OVNKubernetes:AdminNetworkPolicy Metrics via telemetry

### PR DESCRIPTION
**Pull request by tssurya**
* [ ] I added CHANGELOG entry for this change.
* [ ] No user facing changes, so no entry in CHANGELOG was needed.

Changelog will go into the 4.16 backport PR since that's what it will be introduced first. So after this merges and I open a backport I will add it there. This is very much a user facing change.

### CODE REVIEW

File: Documentation/data-collection.md [hunk 1]
```diff
@@ -475,9 +475,47 @@ data:
     #
     # owners: (@openshift/networking)
     #
-    # ovnkube_master_egress_routing_via_host" informs if the OVN-K cluster's gateway mode is
+    # ovnkube_controller_egress_routing_via_host informs if the OVN-K cluster's gateway mode is
```
Comment (simonpasquier) [inline Documentation/data-collection.md:478]: Let's fix the small inconsistency
```suggestion
    # cluster:ovnkube_controller_egress_routing_via_host:max informs if the OVN-K cluster's gateway mode is
```

File: Documentation/data-collection.md [hunk 2]
```diff
@@ -475,9 +475,47 @@ data:
...
     # `routingViaOVN` (0), `routingViaHost` (1) or invalid (2).
-    - '{__name__="cluster:ovnkube_master_egress_routing_via_host:max"}'
+    - '{__name__="cluster:ovnkube_controller_egress_routing_via_host:max"}'
+    #
+    # owners: (@openshift/networking)
+    #
+    # ovnkube_controller_admin_network_policies_db_objects:max informs the total number of
```
Comment (simonpasquier) [inline Documentation/data-collection.md:484]:
```suggestion
    # cluster:ovnkube_controller_admin_network_policies_db_objects:max informs the total number of
```

File: Documentation/data-collection.md [hunk 3]
```diff
@@ -475,9 +475,47 @@ data:
...
+    # OVN database objects (table_name) owned by admin network policies in the cluster.
+    # Labels:
+    # * `table_name`: `ACL` or `AddressSet`.
+    #
+    - '{__name__="cluster:ovnkube_controller_admin_network_policies_db_objects:max"}'
```
Comment (simonpasquier) [inline Documentation/data-collection.md:489]:
```suggestion
    - '{__name__="cluster:ovnkube_controller_admin_network_policies_db_objects:max",table_name=~"ACL|AddressSet"}'
```

File: Documentation/data-collection.md [hunk 4]
```diff
@@ -475,9 +475,47 @@ data:
...
+    #
+    # owners: (@openshift/networking)
+    #
+    # ovnkube_controller_baseline_admin_network_policies_db_objects:max informs the total number of
+    # OVN database objects (table_name) owned by baseline admin network policies in the cluster.
+    # Labels:
+    # * `table_name`: `ACL` or `AddressSet`.
+    #
+    - '{__name__="cluster:ovnkube_controller_baseline_admin_network_policies_db_objects:max"}'
```
Comment (simonpasquier) [inline Documentation/data-collection.md:498]: same here

File: Documentation/data-collection.md [hunk 5]
```diff
@@ -475,9 +475,47 @@ data:
...
+    #
+    # owners: (@openshift/networking)
+    #
+    # ovnkube_controller_admin_network_policies_rules:max informs the total number of
+    # admin network policy rules in the cluster
+    # Labels:
+    # * `direction`: `ingress` or `egress`.
+    # * `action`: `pass` or `allow` or `deny`.
+    #
+    - '{__name__="cluster:ovnkube_controller_admin_network_policies_rules:max"}'
```
Comment (simonpasquier) [inline Documentation/data-collection.md:508]:
```suggestion
    - '{__name__="cluster:ovnkube_controller_admin_network_policies_rules:max",direction=~"ingress|egress",action=~"pass|allow|deny"}'
```

File: Documentation/data-collection.md [hunk 6]
```diff
@@ -475,9 +475,47 @@ data:
...
+    #
+    # owners: (@openshift/networking)
+    #
+    # ovnkube_controller_baseline_admin_network_policies_rules:max informs the total number of
+    # baseline admin network policy rules in the cluster
+    # Labels:
+    # * `direction`: `ingress` or `egress`.
+    # * `action`: `allow` or `deny`.
+    #
+    - '{__name__="cluster:ovnkube_controller_baseline_admin_network_policies_rules:max"}'
```
Comment (simonpasquier) [inline Documentation/data-collection.md:518]: same here

**Comment by tssurya** [comment]
thank you so much Simon for the review!

File: Documentation/data-collection.md (hunk 2 above)
Comment (tssurya) [reply Documentation/data-collection.md:484]: done.

File: Documentation/data-collection.md (hunk 3 above)
Comment (tssurya) [reply Documentation/data-collection.md:489]: done.

File: Documentation/data-collection.md (hunk 4 above)
Comment (tssurya) [reply Documentation/data-collection.md:498]: done.

File: Documentation/data-collection.md (hunk 5 above)
Comment (tssurya) [reply Documentation/data-collection.md:508]: done.

File: Documentation/data-collection.md (hunk 6 above)
Comment (tssurya) [reply Documentation/data-collection.md:518]: done.

**Review by simonpasquier — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**